pkg/clusters/addons/kong/kong.go

```diff
@@ -296,3 +360,75 @@ func urlForService(ctx context.Context, cluster clusters.Cluster, nsn types.Name
 
 	return nil, fmt.Errorf("service %s has not yet been provisoned", service.Name)
 }
+
+func deployKongEnterpriseLicenseSecret(ctx context.Context, cluster clusters.Cluster, namespace, name string) error {
+	license := os.Getenv("KONG_ENTERPRISE_LICENSE")
+	if license == "" {
+		return fmt.Errorf("failed kong enterprise license from environment setting")
+	}
+
+	newSecret := &corev1.Secret{
+		Type: corev1.SecretTypeOpaque,
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Data: map[string][]byte{
+			"license": []byte(license),
+		},
+	}
+
+	_, err := cluster.Client().CoreV1().Secrets(namespace).Create(ctx, newSecret, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed creating kong-enterprise-license secret, err %w", err)
+	}
+
+	fmt.Printf("successfully deployed kong-enterprise-license into the cluster .")
+	return nil
+}
+
+func prepareSecrets(ctx context.Context, namespace string) error {
+	stderr := new(bytes.Buffer)
+	pwd := fmt.Sprintf("--from-literal=password=%s", EnterpriseKongAdminDefaultPWD)
+	cmd := exec.CommandContext(ctx, "kubectl", "create", "secret", "generic", EnterpriseAdminPasswordSecretName, "-n", namespace, pwd)
+	cmd.Stdout = io.Discard
+	cmd.Stderr = stderr
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("failed creating super-admin secret %s: %w", stderr.String(), err)
+	}
+	fmt.Printf("successfully created kong admin secret.")
+
+	pwd, err := os.Getwd()
+	if err != nil {
+		return fmt.Errorf("failed getting current dir, err %w", err)
+	}
+
+	guiF := pwd + "/secret_conf"
+
+	fi, err := os.Create(guiF)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := fi.Close(); err != nil {
+			fmt.Println("failed closing file %w", err)
+			return
+		}
+	}()
+
+	err = ioutil.WriteFile(guiF, []byte(`{"cookie_name":"04tm34l","secret":"change-this-secret","cookie_secure":false,"storage":"kong"}`), 0600)
```

Comment:
Question: I understand that this is a testing environment, but wouldn't it be very straightforward to generate this value for each instance?

Reply:
It is generating this value for each instance.

Reply:
Is it changed somewhere further down the line? Looking at this alone it looks like it's going to use a static `cookie_name` and `secret` value.

That said, IMO a non-issue for now--it is decidedly insecure to do so, but we shouldn't need to care for the test env. IIRC the need for it here is more that Kong fails to start unless you place _some_ value here, so while we don't actually test anything involving the GUI, we need to stuff some value here to let the instance come up. An insecure static value is sufficient for what we need at the moment.

Reply:
Mentioned this to Shane yesterday also, this is exactly following documentation, https://docs.konghq.com/enterprise/2.5.x/deployment/installation/kong-on-kubernetes/#step-5-prepare-the-sessions-plugin-for-kong-manager-and-dev-portal,  pick static value and auth false since we do not use GUI stuff atm.